tools/dump_tool/dump_tool.c

```diff
@@ -389,28 +454,97 @@ int main(int argc, char *argv[])
 		goto tty_lock_err;
 	}
 
-	if (do_handshake(dev_fd) != 0) {
+	printf("Target device locked and ready to DUMP!!!\n");
+
+	printf("Chose from the following options:-\n1. RAM Dump\n2. Userfs Dump\n3. Both Dumps\n4. Exit Dump Tool\n");
+	fflush(stdout);
+	scanf("%d", &choice);
+	getchar();
+
+	if (choice == 2) {
+		printf("Skipping RAM Dump\n");
+		fflush(stdout);
+		goto FS_DUMP;
+	}
+	if (choice == 4) {
+		printf("Skipping both dumps\n");
+		strncpy(handshake_string, EXIT_HANDSHAKE_STRING, HANDSHAKE_STR_LEN_MAX);
+		if (do_handshake(dev_fd, handshake_string) != 0) {
+			printf("Failed to complete handshake for exiting Dump Tool\n");
+			goto handshake_err;
+		}
+		goto TOOL_EXIT;
+	}
+
+	choice--;
+	printf("DUMPING RAM CONTENTS\n");
+	/* Handshake for synchronization before RAMDUMP */
+	strncpy(handshake_string, RAMDUMP_HANDSHAKE_STRING, HANDSHAKE_STR_LEN_MAX);
+	if (do_handshake(dev_fd, handshake_string) != 0) {
 		printf("Target Handshake Failed\n");
 		goto handshake_err;
 	}
 
-	printf("Target entered to ramdump mode\n");
-
 	if (ramdump_info_init(dev_fd) != 0) {
 		printf("Ramdump initialization failed\n");
 		goto init_err;
 	}
 
 	if (ramdump_recv(dev_fd) != 0) {
 		printf("Ramdump receive failed\n");
-		goto ramdump_err;
+		goto init_err;
 	}
 
 	printf("Ramdump received successfully..!\n");
+	getchar();
+
+	ret = read(dev_fd, &c, 1);
+	if (ret != 1) {
+		printf("Failed to get synchronization character from device, ret: %d\n", ret);
+		goto init_err;
+	}
+
+	if (c != 'F') {
```

Comment:
I don't know why this is necessary. You always send 'F' after RAMDUMP.

Reply:
The intention was to check that no characters transmitted from the TARGET are remaining and yet to be read.
This is for synchronization.

Reply:
Then, what's the difference between this and handshake? One of handshake purpose is synchronization. It waits first character to sync.

Reply:
Updated,
Not added with new version.